index.js

```diff
@@ -51,15 +51,131 @@ what's on disk?
+      console.log(path)
+      // Write the file to the FS
+      responseJSON.artifacts.forEach((image, num) => {
+        fs.writeFileSync(
+          path,
+          Buffer.from(image.base64, 'base64')
+        );
+      });
+    })();
+}
+
+// Get story in text from openAI using the text as a prompt
+// The directooryPath is where to save the files that are later generated
+async function getTextStory(text, directoryPath) {
+    const axios = require('axios');
+    const apiEndpoint = 'https://api.openai.com/v1/chat/completions';
+    // Will take out the api key
+    const apiKey = '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';
+
+    const headers = {
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${apiKey}`
+    };
+
+    const data = {
+        "model": "gpt-3.5-turbo",
+        "temperature": 0.1,
+        'messages': [
+          {'role': 'system', 'content': 'I am going to draw an illustrated book of the origins story of ' + text + '. Tell me what to draw assuming I have no prior knowledge in 5 sentences with each sentence describing one element from the book. Omit the word young.'},
+        ]
+    };
+    try {
+      // Make the request
+        const response = await axios.post(apiEndpoint, data, { headers });
+        console.log(response.data.choices[0].message.content);
+        const text = response.data.choices[0].message.content;
+
+        // Parse the comments into an array based on numbering - sometimes the output is in a numbered list
+        filteredSentences = text.split(/\d+\.\s+/).filter(sentence => sentence.trim() !== "");
+
+        // If the output is not a numbered list, there will only be one element in the array that we still need to parse
+        // because it will still be a whole paragraph. Parse based periods, etc.
+        if(filteredSentences.length == 1) {
+          const paragraph = filteredSentences[0];
+          filteredSentences = paragraph.split(/[.!?]+/).filter(sentence => sentence.trim() !== "");
+        }
+
+        // Output the parsed sentences with its index and value
+        for (let i = 0; i < filteredSentences.length; i++) {
+          console.log(`Index: ${i}, Value: ${filteredSentences[i]}`);
+          // Generate an image for this sentence
+          // Pass in index to maintain ordering within the storyboard
+          generateImage(filteredSentences[i], i, directoryPath);
+        }
+    } catch (error) {
+        console.error(error);
+    }
+}
+
 app.post('/generate', (req, res) => {
     const {character} = req.body;
-    console.log(character)
-
-    // call gpt api to get title and scripts
-
-    // call stability ai api to get images
-
-    // save the image respond and other info to the disk folder
+    console.log("The input is " + character)
+    const fs = require('fs');
+
+    // Create directory path to store the images
+    const directoryPath = './public/creations/' + character;
```

Comment:
or we can make it a uuid

Reply:
@wuann15 No I think we can use uuid, since front-end is expecting the path to be under public/creations

Reply:
For the demo, we can use the character name as title, but for the same character, we may create different stories. So later it's better to change it something else.